### ISSUE TITLE
Kubernetes can't resolve external hostnames

### ISSUE THREAD

**Issue by jeroenjacobs1205** [kubernetes/kubernetes#49113]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
 /kind bug
> /kind feature

**What happened**: Kubernetes is unable to resolve external hostnames.

**What you expected to happen**: Kubernetes fails to resolved external names and appends the search domains for some reason.

**How to reproduce it (as minimally and precisely as possible)**:

From within my container, I do an nslookup:

`nslookup www.telenet.be`

This returns an ip address that is associated with a wildcard cname record, instead of the correct ip address.

This returns an incorrect ip adres. I enabled query logging in the dnsmasq container of kubedns, and this what I get back:

```
I0718 11:44:24.488291       1 nanny.go:108] dnsmasq[12]: query[A] www.telenet.be.kring-tst-02.svc.cluster.local from 10.42.0.8
I0718 11:44:24.488954       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.kring-tst-02.svc.cluster.local to 127.0.0.1
I0718 11:44:24.488978       1 nanny.go:108] dnsmasq[12]: query[AAAA] www.telenet.be.kring-tst-02.svc.cluster.local from 10.42.0.8
I0718 11:44:24.488989       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.kring-tst-02.svc.cluster.local to 127.0.0.1
I0718 11:44:24.489303       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.kring-tst-02.svc.cluster.local is NXDOMAIN
I0718 11:44:24.489328       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.kring-tst-02.svc.cluster.local is NXDOMAIN
I0718 11:44:24.489562       1 nanny.go:108] dnsmasq[12]: query[A] www.telenet.be.svc.cluster.local from 10.42.0.8
I0718 11:44:24.490145       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.svc.cluster.local to 127.0.0.1
I0718 11:44:24.490173       1 nanny.go:108] dnsmasq[12]: query[AAAA] www.telenet.be.svc.cluster.local from 10.42.0.8
I0718 11:44:24.490184       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.svc.cluster.local to 127.0.0.1
I0718 11:44:24.490458       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.svc.cluster.local is NXDOMAIN
I0718 11:44:24.490482       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.svc.cluster.local is NXDOMAIN
I0718 11:44:24.491173       1 nanny.go:108] dnsmasq[12]: query[A] www.telenet.be.cluster.local from 10.42.0.8
I0718 11:44:24.491193       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.cluster.local to 127.0.0.1
I0718 11:44:24.491204       1 nanny.go:108] dnsmasq[12]: query[AAAA] www.telenet.be.cluster.local from 10.42.0.8
I0718 11:44:24.491214       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.cluster.local to 127.0.0.1
I0718 11:44:24.491467       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.cluster.local is NXDOMAIN
I0718 11:44:24.491491       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.cluster.local is NXDOMAIN
I0718 11:44:24.491709       1 nanny.go:108] dnsmasq[12]: query[A] www.telenet.be.tst01.kring.mycompany.com from 10.42.0.8
I0718 11:44:24.491889       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.tst01.kring.mycompany.com to 10.9.0.1
I0718 11:44:24.491906       1 nanny.go:108] dnsmasq[12]: query[AAAA] www.telenet.be.tst01.kring.mycompany.com from 10.42.0.8
I0718 11:44:24.492158       1 nanny.go:108] dnsmasq[12]: forwarded www.telenet.be.tst01.kring.mycompany.com to 10.9.0.1
I0718 11:44:24.496889       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.tst01.kring.mycompany.com is <CNAME>
I0718 11:44:24.496928       1 nanny.go:108] dnsmasq[12]: reply lb-app.tst01.kring.mycompany.com is 10.9.130.57
I0718 11:44:24.497035       1 nanny.go:108] dnsmasq[12]: reply www.telenet.be.tst01.kring.mycompany.com is <CNAME>
```

When I open a shell in the container, and modify the following line in resolv.conf

`ndots:5`

to:

`ndots:1`

external names resolves fine again.

So basically, this setting breaks all external dns queries, so this qualifies as a serious bug in my opinion.

So please, tell me where and how I can fix this for all my pods.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.6.6
- Cloud provider or hardware configuration**: VSphere
- OS (e.g. from /etc/os-release): Centos 7
- Kernel (e.g. `uname -a`): Linux worker-002 3.10.0-514.10.2.el7.x86_64 #1 SMP Fri Mar 3 00:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubeadm
- Others:

**Comment by wangwu50** [comment]
I have the same issue .
OS:ubuntu16.04
Install tools: kubeadm
Kubernetes version (use kubectl version): 1.7.2

**Comment by m1093782566** [comment]
@jeroenjacobs1205 Please refer #33554 - there was a lot of discussions.

**Comment by thockin** [comment]
dup of #33554
